db/model/botData.js

```diff
@@ -62,6 +64,29 @@ module.exports = function botData(seq, dataTypes) {
     hooks: {
 
       /**
+       * If the botId is a bot name is validates it and updates the
+       * botId with the actual ID.
+       *
+       * @param {Aspect} inst - The instance being validated
+       * @returns {undefined} - OK
+       */
+      beforeValidate(inst /* , opts */) {
+        const botId = inst.getDataValue('botId');
+        if (looksLikeId.test(botId)) {
+          return seq.Promise.resolve(inst);
+        }
+
+        return seq.models.Bot.findOne({
+          where: {
+            name: botId,
```

Comment:
since this isn't a id, then it's a name -- are you intentionally choosing to do a case sensitive find here? if so, please add comment saying that